app/invocation/invocation_share_button.tsx

```diff
@@ -42,12 +45,33 @@ export default class InvocationShareButtonComponent extends React.Component<
 
   private inputRef = React.createRef<HTMLInputElement>();
 
+  componentWillMount() {
+    let handle = shortcuts.register(KeyCombo.shift_c, () => {
+      this.copyShareUrl();
+    });
+    this.state.keyboardShortcutHandle = handle;
```

Comment:
Setting things on state like `this.state.<...> = <...>` without calling `setState()` is generally  an anti-pattern in React.

Another option is to change the `state` initializer at the top of the class:

```
state = {
  keyboardShortcutHandle: shortcuts.register(KeyCombo.shift_c, () => this.copyShareUrl()),
  ...this.getInitialState(),
}
```

Reply:
Whoops, thought I cleaned all these up.